app/tests/unit/components/Form/ProjectReportingRequirementsForm.test.tsx

```diff
@@ -0,0 +1,243 @@
+    removeButton.click();
+    expect(
+      componentTestingHelper.environment.mock.getMostRecentOperation().request
+    ).toMatchObject({
+      variables: {
+        input: {
+          id: "mock-project-quarterly-report-form-2",
+        },
+        connections: expect.any(Array),
+      },
+    });
+  });
+
+  it("calls useMutationWithErrorMessage and returns expected message when the remove button is clicked", () => {
+    componentTestingHelper.loadQuery();
+    componentTestingHelper.renderComponent();
+
+    const removeButton = screen.getAllByText("Remove")[0];
+    removeButton.click();
+    act(() => {
+      componentTestingHelper.environment.mock.rejectMostRecentOperation(
+        new Error()
+      );
+    });
+
+    expect(componentTestingHelper.errorContext.setError).toBeCalledWith(
+      "An error occurred when deleting."
+    );
+  });
+
+  it("Validates all contact forms when the submit button is clicked", () => {
+    componentTestingHelper.loadQuery();
+    componentTestingHelper.renderComponent();
+
+    const validateFormWithErrors = jest.spyOn(
+      require("lib/helpers/validateFormWithErrors"),
+      "default"
+    );
+
+    screen.getByText(/Submit/i).click();
+    // Once per form
+    expect(validateFormWithErrors).toHaveBeenCalledTimes(2);
+  });
+
+  it("stages the form changes when the `submit` button is clicked", () => {
+    componentTestingHelper.loadQuery();
+    componentTestingHelper.renderComponent();
+    screen.getByText(/submit/i).click();
+    expect(
+      componentTestingHelper.environment.mock.getMostRecentOperation().request
+        .variables.input
+    ).toMatchObject({
+      formChangePatch: { changeStatus: "staged" },
+    });
+  });
+
+  it("reverts the form_change status to 'pending' when editing", async () => {
+    const mockResolver = {
+      ...defaultMockResolver,
+      ProjectRevision(context, generateID) {
+        const result: Partial<ProjectQuarterlyReportForm_projectRevision> = {
+          projectQuarterlyReportFormChanges: {
+            edges: [
+              {
+                node: {
+                  id: `mock-project-quarterly-report-form-${generateID()}`,
+                  newFormData: {
+                    status: "on_track",
+                    dueDate: "2022-01-01",
+                    projectId: 51,
+                    reportType: "Quarterly",
+                  },
+                  operation: "CREATE",
+                  changeStatus: "staged",
+                  formChangeByPreviousFormChangeId: null,
+                },
+              },
+            ],
+            __id: "client:mock:__connection_projectQuarterlyReportFormChanges_connection",
+          },
+        };
+        return result;
+      },
+    };
+    componentTestingHelper.loadQuery(mockResolver);
+    componentTestingHelper.renderComponent();
+
+    userEvent.type(screen.getByLabelText(/General Comments \(optional\)/), "c");
```

Comment:
If I tried to type anything longer than one character, letters would be missing in the `expect`. I tried a few versions of wrapping things in `act()` but no dice

Reply:
I think it's because the mutation fails to return complete data (we can confirm that on Monday) and the optimistic update might get in the way. So on each character type, the change event is fired and the react component re-renders with the initial, empty state from the mocked resolver.

As an alternative, you can artificially fire a `change` event on the input:
```tsx
    fireEvent.change(screen.getByLabelText(/General Comments \(optional\)/), {
      target: { value: "some test comment" },
    });
```

Reply:
Yes, our change to the optimistic response solved this